drivers/modem/gsm_ppp.c

```diff
@@ -349,6 +349,12 @@ static void set_ppp_carrier_on(struct gsm_modem *gsm)
 static void gsm_finalize_connection(struct gsm_modem *gsm)
 {
 	int ret;
+	static int attach_retry;
```

Comment:
Could you add this retry counter to `gsm_modem` context, as then we could (at least in theory) have multiple modem devices in the system.